dali/imgcodec/util/convert.h

```diff
@@ -67,6 +69,54 @@ void Convert(Out *out, const int64_t *out_strides,
   }
 }
 
+/**
+ * @brief A functor for converting between color spaces.
+ *
+ * It reads the input data from a tensor, passes it to user-provided conversion function, and
+ * then stores the result. Both the argument and return value of the conversion function can
+ * either be a scalar or a vector.
+ */
+template <typename FuncIn, typename FuncOut>
+struct ConvertColorSpace {
+  using FuncType = FuncOut(*)(FuncIn);
+
+  using InVec = typename std::conditional<is_vec<FuncIn>::value, FuncIn, vec<1, FuncIn>>::type;
+  using OutVec = typename std::conditional<is_vec<FuncOut>::value, FuncOut, vec<1, FuncOut>>::type;
+  using In = typename InVec::element_t;
+  using Out = typename OutVec::element_t;
+
+  void load(In& target, const In *in) {
```

Comment:
I think it would be better to just do:
```suggestion
  static constexpr In load(const In *in) { return *in; }
```

Reply:
This won't be that easy, because then we would have two overloads with same arguments and it won't compile. I added some `enable_if` magic to solve that, I'd be grateful for feedback on that